File: January 2023/Day 23/Geeks And The String/geeks-and-the-string.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    string removePair(string s) {
        // code here
        string ans = "";
        for(int i=0;i<s.length();i++) {
            char c = s[i];
            if(ans.length()>0 && ans[ans.length()-1]==c) {
                ans = ans.substr(0,ans.length()-1);
            } else {
                ans+=c;
            }
        }
        if(ans.length()==0) return "-1";
        return ans;
    }
};


//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        string s; 
        cin>>s;
        
        Solution obj;
        string res = obj.removePair(s);
        
        cout<<res<<endl;
        
    }
}

// } Driver Code Ends

File: January 2023/Day 23/Geeks And The String/geeks-and-the-string.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t;
        t = Integer.parseInt(br.readLine());
        while(t-- > 0){
            
            String s; 
            s = br.readLine();
            
            Solution obj = new Solution();
            String res = obj.removePair(s);
            
            System.out.println(res);
            
        }
    }
}

// } Driver Code Ends


class Solution {
    public static String removePair(String s) {
        // code here
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<s.length();i++) {
            char c = s.charAt(i);
            if(stack.size()>0 && stack.peek()==c) {
                stack.pop();
            } else {
                stack.push(c);
            }
        }
        if(stack.size()==0) return "-1";
        StringBuilder sb = new StringBuilder();
        while(stack.size()>0) sb.append(Character.toString(stack.pop()));
        return sb.reverse().toString();
    }
}
        
